src/lib/loaders/loaders_with_authentication/gravity.ts

```diff
@@ -561,6 +561,15 @@ export default (accessToken, userID, opts) => {
       ({ partnerId, userId }) =>
         `partner_collector_profile?partner_id=${partnerId}&user_id=${userId}`
     ),
+    partnerCollectorProfileUserInterestsLoader: gravityLoader<
+      any,
+      { collectorProfileId: string; partnerId: string }
+    >(
+      ({ collectorProfileId, partnerId }) =>
+        `partner_collector_profile/${collectorProfileId}/user_interests?partner_id=${partnerId}`,
```

Comment:
**thought:** I didn't notice this in the Gravity PR, but it feels strange to pass a partner ID to this endpoint. Instead of passing a partner ID, I think Gravity can figure out whether the current user has access to a partner by doing:

```ruby
error!('Forbidden', 403) unless current_user.has_authorization_to?(:update, partner)
```

Reply:
Was a decision we took back there when pairing with @jonallured , I remember we discussed about but don't recall exactly why we didn't use the `current_user`.

Reply:
I might be missing something but should we do both? A user can have access to multiple partners so I would expect for an ID to be passed and then also a check in the endpoint to verify whether the user can see that partner.

Reply:
Good point. I forgot that Gravity will need to fetch the `partner` in that snippet.